Explorer/Assets/Scripts/ECS/Unity/PrimitiveColliders/Systems/InstantiatePrimitiveColliderSystem.cs

```diff
@@ -0,0 +1,135 @@
+            { PBMeshCollider.MeshOneofCase.Plane, new SetupPlaneCollider() },
+        };
+
+        private readonly Dictionary<PBMeshCollider.MeshOneofCase, ISetupCollider> setupColliderCases;
+
+        internal InstantiatePrimitiveColliderSystem(World world, IComponentPoolsRegistry poolsRegistry,
+            Dictionary<PBMeshCollider.MeshOneofCase, ISetupCollider> setupColliderCases = null) : base(world)
+        {
+            this.setupColliderCases = setupColliderCases ?? SETUP_COLLIDER_LOGIC;
+            this.poolsRegistry = poolsRegistry;
+        }
+
+        protected override void Update(float t)
+        {
+            InstantiateNonExistingColliderQuery(World);
+            TrySetupExistingColliderQuery(World);
+        }
+
+        [Query]
+        [All(typeof(PBMeshCollider), typeof(SDKTransform), typeof(Transform))]
+        [None(typeof(PrimitiveColliderComponent))]
+        private void InstantiateNonExistingCollider(in Entity entity, ref PBMeshCollider sdkComponent, ref Transform transform)
+        {
+            var component = new PrimitiveColliderComponent();
+            Instantiate(setupColliderCases[sdkComponent.MeshCase], ref component, ref sdkComponent, ref transform);
+            World.Add(entity, component);
+        }
+
+        [Query]
+        [All(typeof(PBMeshCollider), typeof(SDKTransform), typeof(Transform), typeof(PrimitiveColliderComponent))]
+        private void TrySetupExistingCollider(
+            ref PrimitiveColliderComponent primitiveColliderComponent,
+            ref PBMeshCollider sdkComponent,
+            ref Transform transform)
+        {
+            if (!sdkComponent.IsDirty) return;
+
+            ISetupCollider setupCollider = setupColliderCases[sdkComponent.MeshCase];
+
+            // Prevent calling an overloaded comparison from UnityEngine.Object
+            if (ReferenceEquals(primitiveColliderComponent.Collider, null))
+                Instantiate(setupCollider, ref primitiveColliderComponent, ref sdkComponent, ref transform);
+            else
+
+                // Just a change of parameters
+                SetupCollider(setupCollider, primitiveColliderComponent.Collider, sdkComponent);
+
+            sdkComponent.IsDirty = false;
+        }
+
+        private void SetupCollider(ISetupCollider setupCollider, Collider collider, in PBMeshCollider sdkComponent)
+        {
+            // Setup collider only if it's gonna be enabled, otherwise there is no reason to [re]generate a shape
+            if (SetColliderLayer(collider, sdkComponent))
+                setupCollider.Execute(collider, sdkComponent);
+        }
+
+        private bool SetColliderLayer(Collider collider, in PBMeshCollider sdkComponent)
+        {
+            uint colliderLayer = sdkComponent.GetColliderLayer();
+
+            GameObject colliderGameObject = collider.gameObject;
+
+            var enabled = true;
+
+            if ((colliderLayer & LAYER_PHYSICS_POINTER) == LAYER_PHYSICS_POINTER)
```

Comment:
This if is a little bit confusing. To confirm, what its saying is that, if no layer is set, then the collider should not be enabled?

Reply:
This is how it works in the current renderer, I believe it's a fallback path. Essentially it means that no collisions/ raycasting will be available, therefore no reason to keep the collider in the physics world